src/main/java/janggi/board/Board.java

```diff
@@ -26,71 +29,96 @@ public static Board createInitializedJanggiBoard() {
         return new Board(initMap);
     }
 
-    public void updatePosition(final Position source, final Position destination, final Country country) {
-        validatePositionAndTeam(source, country);
+    public void updatePosition(final JanggiPosition source, final JanggiPosition destination, final Country currentTurnCountry) {
+        validateExistsPieceInPosition(source);
+        validateIsCurrentCountry(source, currentTurnCountry);
         validatePieceCanMove(source, destination);
 
         movePieceToDestination(source, destination);
     }
 
-    private void validatePositionAndTeam(final Position source, final Country country) {
-        if (!janggiBoard.containsKey(source) || !janggiBoard.get(source).equalsTeamType(country)) {
-            throw new IllegalArgumentException("scr 좌표에 기물이 존재하지 않거나, 해당 팀의 기물이 아닙니다.");
+    private void validateExistsPieceInPosition(final JanggiPosition source) {
+        if (!janggiBoard.containsKey(source)) {
+            throw new IllegalArgumentException("scr 좌표에 기물이 존재하지 않습니다.");
+        }
+    }
+
+    private void validateIsCurrentCountry(final JanggiPosition source, final Country country) {
+        if (!janggiBoard.get(source).equalsCountry(country)) {
+            throw new IllegalArgumentException("현재 턴에 해당하는 기물이 아닙니다.");
         }
     }
 
-    private void validatePieceCanMove(final Position source, final Position destination) {
+    private void validatePieceCanMove(final JanggiPosition source, final JanggiPosition destination) {
         final Piece piece = janggiBoard.get(source);
 
         if (!piece.isAbleToMove(source, destination, this)) {
             throw new IllegalArgumentException("해당 기물은 해당 위치로 이동할 수 없습니다.");
         }
     }
 
-    private void movePieceToDestination(final Position source, final Position destination) {
+    private void movePieceToDestination(final JanggiPosition source, final JanggiPosition destination) {
         janggiBoard.put(destination, janggiBoard.get(source));
         janggiBoard.remove(source);
     }
 
+    public boolean isAliveAllGenerals() {
+        return janggiBoard.values().stream()
+                .filter(Piece::isGeneral)
+                .count() == ALL_GENERAL_COUNT;
+    }
+
+    public JanggiScore calculateScoreByCountry(final Country country) {
+        final Pieces piecesByCountry = findAllByCountry(country.toggleCountry());
+        return piecesByCountry.calculateAllScoreByCountry(country);
+    }
```

Comment:
인자를 통해서 이미 Country로 점수를 계산한다는 것을 드러내고 있는 것 같습니다.

```suggestion
    public JanggiScore calculateScore(final Country country) {
        final Pieces piecesByCountry = findAllByCountry(country.toggleCountry());
        return piecesByCountry.calculateAllScoreByCountry(country);
    }
```

Reply:
이미 적국을 기반으로 Pieces를 추려냈는데,
전체 점수를 계산할 때 Country가 왜 필요한가요?

Reply:
Country 마다 최대 점수가 달라서 이를 구분해줄 필요가 있었습니다.